2025/lab/Keys/Week7_Missingness_Robustness.Rmd

```diff
@@ -0,0 +1,551 @@
+```
+
+The column `meth` tells us that collinearity is to blame for our problem variables. We can then select these columns out to create a new dataset that doesn't include the variables we don't want in our final regression. (We'll keep `logincome` since we're removing `income`.) I also set `printFlag = T` so that we don't get the output from above. 
+
+```{r mice2}
+newimp <- gss %>% 
+  select(-wrkslf, -workexp, -income, -imp_loginc, -imp_income, -sex) %>% 
+  mice(., m= 5, maxit = 2,
+       seed = seed, printFlag=F)
+newimp$loggedEvents
+
+```
+
+The `NULL` output when we ask for logged events means there were no issues here. 
+
+### MICE Diagnostics
+
+There are several ways to evaluate our imputation. One diagnostic plot we can make is to map the densities of our imputed variables (in red) against the actual (in blue). This might remind you of the posterior predictive checks we use for Bayesian models. 
+
+```{r mice_diag}
+densityplot(newimp, layout = c(3,3))
+```
+
+These suggest that certain variables like `workhrs` and `unemp` are well imputed by relationships with the other variables, while `educ` and `prestige` have difficulty. A way to improve the imputations would be to specify which variables should be used to predict the others, but that's beyond our scope for today. For example, saying that $agesq = age^2$ would greatly improve prediction of `agesq`. Even better would be to create `agesq` *after* imputation. 
+
+### Modeling with Imputed Data
+
+To create a linear regression with imputed data, we cannot use our normal function. Instead we use the `with()` function to tell R to regress "with" our imputed dataset. (As an aside, we could've always used `with(data, lm(y ~x))`, but it didn't make much sense.) 
+
+```{r imp-reg}
+# CAN'T USE THIS WITH MICE DATA
+# impmod1 <- lm(imp_loginc ~ prestige + self_employed + race + female, data = newimp)
+
+impmod1 <- with(newimp, lm(logincome ~ prestige + self_employed + race + female))
+
+summary(impmod1)
+```
+
+What's this?! Remember that multiple imputation gives you multiple datasets, so now R ran a separate regression on each of these datasets.
+
+If you simply run `summary` on the regression object, R will give you one row for each term, for each dataset.
+
+Instead, we can use `mice::pool()` to take the average of the parameters. 
+
+```{r imp-sum}
+
+impmod_pool <- pool(impmod1)
+summary(impmod_pool)
+
+```
+
+The `modelsummary()` command will run this for us and will pool the R-squared, but it won't give us other goodness-of-fit info, including the number of observations and how many imputed datasets we have. For this reason, it's generally better to pool your models first. 
+
+```{r imp_msumm}
+
+modelsummary(impmod_pool, estimate = "{estimate} ({std.error}) {stars}",
+             statistic = NULL,
+             output = "huxtable")
+
+```
+
+To explore how the number of imputed datasets can affect the output, let's create one with 100 imputations and 5 maximum iterations. 
```

Comment:
@brenthoagland This part with 100 imputations seems like overkill. I think you can safely drop it (i.e. delete remainder of this section).